sim/paladin/righteous_vengeance.go

```diff
@@ -0,0 +1,72 @@
+package paladin
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+)
+
+const RVTicks = 4
+
+func (paladin *Paladin) registerRV() {
+	if !paladin.hasRune(proto.PaladinRune_RuneCloakRighteousVengeance) {
+		return
+	}
+
+	paladin.RegisterAura(core.Aura{
+		Label:    "Righteous Vengeance",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+
+			if spell.Flags.Matches(SpellFlag_RV) && result.DidCrit() {
+				paladin.procRV(sim, result)
+			}
+		},
+	})
+
+	paladin.rv = paladin.RegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: 440675},
+		SpellSchool: core.SpellSchoolHoly,
+		//DefenseType: core.DefenseTypeMagic,
+		ProcMask:    core.ProcMaskEmpty,  //Not sure
```

Comment:
As long as the RV dot ticks don't proc effects then this is correct

Reply:
I'm not sure if it can proc spell/dot procs, it does have the suppress weapon procs flag though. What proc mask would be best here?

Reply:
Fixed , though we don't really have many important procs that would proc here I think